### ISSUE TITLE
Write EEG/iEEG reference to JSON sidecar

### ISSUE THREAD

**Issue by hoechenberger** [mne-tools/mne-bids#713]
See https://github.com/mne-tools/mne-bids/pull/695#issuecomment-786165997

**Comment by sappelhoff** [comment]
> amend our documentation, suggesting users do the following before calling write_raw_bids:
> - if the reference channels are present in the data, call raw.set_eeg_reference() to make the reference explicitly known
> - if the reference channels are not present in the data:
>    - add them via raw.add_reference_channels()
>    - call raw.set_eeg_reference() to make the reference explicit
> The cool thing would be that if the reference electrode is part of the montage, we'd get the reference electrode location for free :)
Then call write_raw_bids(), which should extract the reference channel info from raw.info['chs'] -> this will need a PR
Easiest is to start with the assertion that all channels have the same reference, so we can write it to [i]EEGReference in the sidecar JSON.


and if users don't do it we keep writing `n/a`?

**Comment by hoechenberger** [comment]
> and if users don't do it we keep writing `n/a`?

Yeah… I mean, if you will, this follows the principle we're applying to events already: if they are not specified, we don't write any (well… we do warn, though…)

**Comment by hoechenberger** [comment]
I'll make a PR and we can then discuss & iterate from there

**Comment by hoechenberger** [comment]
Oh well it wouldn't be fun it it were easy, right?
https://github.com/mne-tools/mne-python/issues/8962
